src/main/java/gregtech/api/recipes/ModHandler.java

```diff
@@ -278,6 +278,25 @@ public static void addShapedEnergyTransferRecipe(String regName, ItemStack resul
         ForgeRegistries.RECIPES.register(shapedOreRecipe);
     }
 
+    public static void addShapedEnergyTransferRecipe(String regName, ItemStack result, Predicate<ItemStack> chargePredicate, long chargeAmount, boolean transferMaxCharge, Object... recipe) {
+        boolean skip = false;
+        if (result.isEmpty()) {
+            GTLog.logger.error("Result cannot be an empty ItemStack. Recipe: {}", regName);
+            GTLog.logger.error("Stacktrace:", new IllegalArgumentException());
+            skip = true;
+        }
+        skip |= validateRecipe(regName, recipe);
+        if (skip) {
+            RecipeMap.setFoundInvalidRecipe(true);
+            return;
+        }
+
+        IRecipe shapedOreRecipe = new ShapedOreEnergyTransferRecipe(null, result.copy(), chargePredicate, chargeAmount, transferMaxCharge, finalizeShapedRecipeInput(recipe))
+            .setMirrored(false) //make all recipes not mirrored by default
+            .setRegistryName(regName);
+        ForgeRegistries.RECIPES.register(shapedOreRecipe);
+    }
+
```

Comment:
This is nearly 100% code duplication except one param in method header and then passing it. This could be done by extending current one by that param and its passing down. And creating new function with old param set which just calls that updated one.

Reply:
I have done so